### PR TITLE
Added ALTER COLUMN DEFAULT

### PR DESCRIPTION
**Pull request by Hydrocharged**

### CODE REVIEW

File: enginetest/script_queries.go [hunk 1]
```diff
@@ -621,4 +621,52 @@ var ScriptTests = []ScriptTest{
...
+				Query:       "ALTER TABLE test ALTER v2 SET DEFAULT 1;",
+				ExpectedErr: sql.ErrTableColumnNotFound,
+			},
+			{
+				Query:    "ALTER TABLE test ALTER v1 DROP DEFAULT;",
+				Expected: []sql.Row{},
+			},
+			{
+				Query:       "INSERT INTO test (pk) VALUES (3);",
+				ExpectedErr: plan.ErrInsertIntoNonNullableDefaultNullColumn,
```
Comment (zachmu) [inline enginetest/script_queries.go:660]: Should move this to the sql package, and make sure the error message matches MySQL

File: sql/core.go [hunk 2]
```diff
@@ -637,6 +637,15 @@ type AlterableTable interface {
 	ModifyColumn(ctx *Context, columnName string, column *Column, order *ColumnOrder) error
 }
 
+// AlterableDefaultsTable should be implemented by tables that can add, change, or remove the default values on their columns.
+type AlterableDefaultsTable interface {
```
Comment (zachmu) [inline sql/core.go:641]: Can't this be represented by AlterableTable.ModifyColumn? If so, would prefer that to another interface

Seems like `ALTER TABLE t1 ALTER col DROP DEFAULT` should be equivalent to `ALTER TABLE t1 MODIFY col <new definition without default>`

Ditto for setting a new default

**Review by zachmu — COMMENTED** [review]
Question about the new interface

File: sql/core.go (hunk 2 above)
Comment (Hydrocharged) [reply sql/core.go:641]: Yeah that’s a fair point, I’ll rework it

**Pull request merged**